asab/library/providers/git.py

```diff
@@ -0,0 +1,145 @@
+import os
+import tempfile
+import logging
+import hashlib
+
+import pygit2
+
+from .filesystem import FileSystemLibraryProvider
+from ...config import Config
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+
+class GitLibraryProvider(FileSystemLibraryProvider):
+	"""
+	Read-only git provider to read from remote repository.
+	It clones a remote git repository to a temporary directory and then uses the
+	FileSystemLibraryProvider to read the files.
+	To read from local git repository, please use FileSystemProvider.
+
+	Configuration:
+	(Use either deploytoken, publickey+privatekey for SSH option, or username and password and HTTP access.)
+
+
+	```
+	[library]
+	providers=git+<URL or deploy token>
+
+	[library:git]
+	publickey=<absolute path to file>
+	privatekey=<absolute path to file>
+	username=johnsmith
+	password=secretpassword
+	repodir=<optional location of the repository cache>
+	```
+	"""
+	def __init__(self, library, path):
+		self.LastCommit = None
+		self.URL = path[4:]
+
+		self.Callbacks = pygit2.RemoteCallbacks(get_git_credentials(self.URL))
+
+		# TODO: Check `repodir`
```

Comment:
I switched this implementation (as part of my review) to a persistent Git library.
I would be cool, if the user can optionally specify the path of the temp folder to store git repos.